src/AggregateRoots/AggregateRoot.php

```diff
@@ -195,12 +195,19 @@ private function apply(ShouldBeStored $event): void
 
         $applyingMethodName = "apply{$camelCasedBaseName}";
 
-        if (method_exists($this, $applyingMethodName)) {
+        $reflectionClass = new ReflectionClass($this);
+
+        $applyMethodExists = $reflectionClass->hasMethod($applyingMethodName);
+        $applyMethodIsPublic = $applyMethodExists && $reflectionClass->getMethod($applyingMethodName)->isPublic();
+
+        if ($applyMethodExists && $applyMethodIsPublic) {
             try {
                 app()->call([$this, $applyingMethodName], ['event' => $event]);
```

Comment:
What's the benefit of using `app()->call()`? Can't we just always do `$this->$applyingMethodName($event);` instead?

Reply:
`app()->call()` allows the package user to inject additional dependencies in the `apply*` methods.

I don't see much value in that feature, but removing it now would be a breaking change.

Reply:
Ah right, forgot about that